pkgs/servers/nosql/mongodb/mongodb.nix

```
@@ -33,7 +33,7 @@ with lib;
 }:
 
 let
-  scons = buildPackages.scons;
+  scons = buildPackages.scons.override{ python3 = python311; };
```

this likely needs to override using `buildPackages.python311`, and the top-level python3 argument isn't even used otherwise.

---

Like this? Seems to produce the same hashes at least on x86_64-linux

```suggestion
  scons = buildPackages.scons.override { python3 = buildPackages.python311; };
```

---

I don't think this comment is correct. This PR seems to fix the build problem as is.